addons/hr_holidays/models/hr_leave.py

```diff
@@ -936,16 +947,17 @@ def write(self, values):
                     else:
                         employees = self.mapped('employee_id')
                     self._check_double_validation_rules(employees, values['state'])
-            if 'date_from' in values:
-                values['request_date_from'] = values['date_from']
-            if 'date_to' in values:
-                values['request_date_to'] = values['date_to']
         result = super(HolidaysRequest, self).write(values)
         if not self.env.context.get('leave_fast_create'):
             for holiday in self:
                 if employee_id:
                     holiday.add_follower(employee_id)
 
+            protected_fields = {'date_from', 'date_to'}
+            date_from_to = protected_fields & set(values)
+
+            self._update_request_date(protected_fields=protected_fields, date_from_to=date_from_to)
+
```

Comment:
It could be simpler :)
```suggestion
        protected_fields = {'date_from', 'date_to'}
        date_from_to = protected_fields & set(values)
       
        if date_from_to:
            with self.env.protecting([self._fields[field] for field in protected_fields], self):
                for leave in self:
                    leave_tz = timezone(leave.tz)
                for date_field in date_from_to:
                    leave['request_' + date_field] = pytz.utc.localize(leave[date_field]).astimezone(leave_tz).date()        
```

Reply:
I think it makes more sense to have the update portion be separated into a function so that other parts can hook into it and add more protected fields if need be. There's definitely customizations on some clients environments that would benefit from being able to hook onto it

Reply:
Given the method stands exclusively to manage dates, it feels too specific IMO, but if you have practical examples I'm open to it.
Additionally, I'm not entirely sure in the first place of the functional process to have date_from/date_to modified rather than request_date_from/request_date_to that are the fields used for the interface... :thinking:

Reply:
@Bertrand2 I did some testing on my end and it seems like you were right and the issue came from the fact that request_date_from/to was being written to twice.

I added a check to make sure that it's only being written if it is different.

For date_from/_to being modified directly, there's plenty of places in the unit test code that writes directly to date_from/to, so I figured it's best to be consistent